packages/core/src/RenderPipeline/RenderQueue.ts

```diff
@@ -198,7 +153,166 @@ export class RenderQueue {
     this.items.length = 0;
   }
 
-  private _isPrimitive(item) {
-    return !!item.mesh;
+  /**
+   * Sort the elements.
+   */
+  sort(isTransparent: boolean = false, left: number = 0, right: number = this.items.length): void {
```

Comment:
A little over-encapsulation,do not need,call this._quickSort and set "compareFunction" is enought. 
and you will add alphaTest RenderQueue soon.